Modules/Common/PolkadotLikeMinimalModule.php

```diff
@@ -0,0 +1,240 @@
+
+    public ?ExtraDataModel $extra_data_model = ExtraDataModel::Default;
+
+    public ?bool $should_return_events = true;
+    public ?bool $should_return_currencies = false;
+    public ?bool $allow_empty_return_events = true;
+
+    public ?bool $mempool_implemented = false;
+    public ?bool $forking_implemented = true;
+
+    //
+
+    public ?array $block_data = null;
+
+    //
+
+    final public function pre_initialize()
+    {
+        $this->version = 1;
+    }
+
+    final public function post_post_initialize()
+    {
+        //
+    }
+
+    final public function inquire_latest_block()
+    {
+        return (int)requester_single($this->select_node(),
+            endpoint: 'api/scan/metadata',
+            params: true,
+            result_in: 'data',
+            timeout: $this->timeout)['blockNum'];
+    }
+
+    final public function ensure_block($block_id, $break_on_first = false)
+    {
+        $this->block_data = requester_single($this->select_node(),
+            endpoint: 'api/scan/block',
+            params: ['block_num' => $block_id],
+            result_in: 'data',
+            timeout: $this->timeout);
+
+        $this->block_hash = $this->block_data['hash'];
+        $this->block_time = date('Y-m-d H:i:s', (int)$this->block_data['block_timestamp']);
+    }
+
+    final public function pre_process_block($block_id)
+    {
+        if ($block_id === 0)
+        {
+            $this->set_return_events([]);
+            return;
+        }
+
+        $extrinsics = $extrinsics_with_hashes = [];
+
+        foreach ($this->block_data['extrinsics'] as $extrinsic)
+            if ($extrinsic['extrinsic_hash'])
+                $extrinsics_with_hashes[] = $extrinsic['extrinsic_hash'];
+
+        if (!$extrinsics_with_hashes)
+        {
+            $this->set_return_events([]);
+            return;
+        }
+
+        $events = [];
+        $multi_curl = [];
+
+        foreach ($extrinsics_with_hashes as $extrinsic_hash)
+            $multi_curl[] = requester_multi_prepare($this->select_node(),
+                endpoint: 'api/scan/extrinsic',
+                params: ['hash' => $extrinsic_hash],
+                timeout: $this->timeout);
+
+        $curl_results = requester_multi($multi_curl, limit: envm($this->module, 'REQUESTER_THREADS'), timeout: $this->timeout);
+
+        foreach ($curl_results as $v)
+        {
+            $extrinsics[] = requester_multi_process($v, result_in: 'data');
+        }
+
+        $transfers = [];
+
+        foreach ($extrinsics as $i)
+        {
+            $this_i = $fee = null;
+
+            if (count($i) !== 1)
+            {
+                $found = false;
+                foreach ($i as $j => $k)
+                {
+                    if ((int)$k['block_num'] === $block_id)
+                    {
+                        if ($found)
+                            throw new ModuleError('Two extrinsics with the same hash in one block');
```

Comment:
This is not an error, this is a documented behaviour for polkadot, there's no need for the logging this as error.